### ISSUE TITLE
Selection background displays over options

### ISSUE THREAD

**Issue by Kommynct** [philj56/tofi#73]
If you go down one with the selection background padding set to -1 it covers up the one above it.

**Comment by philj56** [comment]
You mean with -1 in the same direction as the text, e.g. `selection-background-padding=-1,0` in vertical mode? That's to be expected.

The text gets drawn in order (particularly important in horizontal mode), so the background of a later item will always draw over the text of an earlier item. There's no nice way to fix this, as who says whether you want the selection background to draw in front of behind the other results' backgrounds? Also, I think doing something else would require multiple renders of the text per frame, slowing things down a bit.

The answer is pretty much "don't do that" - you probably only want to set a vertical padding of -1 in horizontal mode, and a horizontal padding of -1 in vertical mode.

**Comment by Kommynct** [comment]
Even if you set a vertical padding of say, 20, it currently displays the background over the text
![image](https://user-images.githubusercontent.com/13646790/205627062-cc9cb3f8-1bb6-4a66-aef9-5250d8519c82.png)

This does seem like a bug, regardless of there being no nice way to fix it that's obvious.

**Comment by philj56** [comment]
Yeah, it kind of depends on how you think about the drawing. If you think "all backgrounds should be behind all text", then this behaviour is definitely wrong. If you think "text and its background are one object", however, this is what you end up with (and that's how tofi handles the drawing). Just like if you e.g. set `result-spacing=-90` or something:

![overlap](https://user-images.githubusercontent.com/13610623/205645796-07e112af-b720-47e8-b6b4-4eb48df226ff.png)

or with a background:

![overlap_background](https://user-images.githubusercontent.com/13610623/205645852-44fdd92b-8ae0-4749-9516-afd2b3b085eb.png)

In this model, the only way to avoid overlaps is to make `*-background-padding` also affect the positioning of all the text (like CSS does):

![no_overlap](https://user-images.githubusercontent.com/13610623/205648915-ab5dc226-67b7-4655-8a95-796cbe2be8a1.png)

However, this then causes the layout to shift around as you change selections (unless you have all the paddings set to be the same, in which case you may as well have just used `result-spacing` anyway).

So yeah, this behaviour can be seen as "buggy", but it's just a consequence of how tofi works. I think I'll label this as a wontfix bug and leave it open for a while, in case someone has some suggestions or strong prefences for how or if this should be handled.

**Comment by Kommynct** [comment]
If this is desired behavior wouldn't you also want it to cover the text below instead of just the text above? That's my only suggestion, other than that I'm good.
